js/main.js

```diff
@@ -129,10 +129,19 @@ function viewMoreDetails(event) {
   }
 
   function getInfo(movieID) {
+    // debugger;
     var xhr2 = new XMLHttpRequest();
     xhr2.open('GET', 'http://www.omdbapi.com/?' + 'i=' + movieID + '&apikey=b1862476&' + '&plot=full');
     xhr2.responseType = 'json';
     xhr2.addEventListener('load', function () {
+      // loader
+      var loaderEl = document.querySelector('.loader');
+      loaderEl.classList.remove('hidden');
+      setTimeout(showPage, 500);
+      function showPage() {
+        loaderEl.classList.add('hidden');
+      }
```

Comment:
You shouldn't do this with a timeout, because it will not match the time it takes to get data back. You should instead hide the loading spinner inside the `load` event handler for the request so that it is hidden as soon as a response is received.